recipes/quickstart/Multi-Modal-RAG/README.md

```diff
@@ -0,0 +1,116 @@
+- Notebook 3: `Part_3_RAG_Setup_and_Validation.ipynb`
+- Script: `final_demo.py`
+
+Here's the detailed outline:
+
+### Step 1: Data Prep and Synthetic Labeling:
+
+[Notebook for Step 1](./notebooks/Part_1_Data_Preperation.ipynb) and [Script for Step 1](./scripts/label_script.py)
+
+To run the script (remember to set n):
+```
+python scripts/label_script.py --hf_token "your_huggingface_token_here" \
+    --input_path "../MM-Demo/images_compressed" \
+    --output_path "../MM-Demo/output/" \
+    --num_gpus N
+```
+
+The dataset consists of 5000 images with some meta-data.
+
+The first half is preparing the dataset for labeling:
+- Clean/Remove corrupt images
+- EDA to understand existing distribution
+- Merging up categories of clothes to reduce complexity 
+- Balancing dataset by randomly sampling images
+
+Second Half consists of Labeling the dataset. We are bound by an interesting constraint here, 11B model can only caption one image at a time:
+- We load a few images and test captioning
+- We run this pipeline on random images and iterate on the prompt till we feel the model is giving good outputs
+- Finally, we can create a script to label all 5000 images on multi-GPU
+
+After running the script on the entire dataset, we have more data cleaning to perform.
+
+### Step 2: Cleaning up Synthetic Labels and preparing the dataset:
+
+[Notebook for Step 2](./notebooks/Part_2_Cleaning_Data_and_DB.ipynb)
+
+Even after our lengthy (apart from other things) prompt, the model still hallucinates categories and label-we need to address this
+
+- Re-balance the dataset by mapping correct categories
+- Fix Descriptions so that we can create a CSV
+
+Now, we are ready to try our vector db pipeline:
+
+### Step 3: Notebook 3: MM-RAG using lance-db to validate idea
+
+[Notebook for Step 3](./notebooks/Part_3_RAG_Setup_and_Validation.ipynb) and [Final Demo Script](./scripts/label_script.py)
+
+
+With the cleaned descriptions and dataset, we can now store these in a vector-db
+
+You will note that we are not using the categorization from our model-this is by design to show how RAG can simplify a lot of things. 
+
+- We create embeddings using the text description of our clothes
+- Use 11-B model to describe the uploaded image
+- Try to find similar or complimentary images based on the upload
+
+We try the approach with different retrieval methods.
+
+Finally, we can bring this all together in a Gradio App. 
+
+For running the script:
+```
+python scripts/final_demo.py \
+    --images_folder "../MM-Demo/compressed_images" \
+    --csv_path "../MM-Demo/final_balanced_sample_dataset.csv" \
+    --table_path "~/.lancedb" \
+    --api_key "your_together_api_key" \
+    --default_model "BAAI/bge-large-en-v1.5" \
+    --use_existing_table 
+```
+
+Task: We can further improve the description prompt. You will notice sometimes the description starts with the title of the cloth which causes in retrieval of "similar" clothes instead of "complementary" items
```

Comment:
maybe Note instead of Task? also would be helpful if you leave an example if its not shown in the notebook/app.